### DISCUSSION TITLE
parallel task scheduling

### DISCUSSION THREAD

**Discussion by schmurfy** [concourse/concourse#7909] [Help & Support]
Hello,
I have been struggling with an issue for a while but still haven't found a satisfying solution, in our pull-requests pipelines we have multiple parallel jobs depending on which applications were modified (mono repo, the pipelines are created on the fly as needed), what almost always happen is that all the tasks in these jobs get scheduled on the same worker, its load skyrocket and the whole pipeline take ages to complete.

I tried tweaking the scheduler with:
```
CONCOURSE_CONTAINER_PLACEMENT_STRATEGY=limit-active-containers,limit-active-volumes,volume-locality,fewest-build-containers
CONCOURSE_MAX_ACTIVE_CONTAINERS_PER_WORKER=150
CONCOURSE_MAX_ACTIVE_VOLUMES_PER_WORKER=100
```

this helped a bit but the same still happens, one poor worker gets chosen and hammered to death with all the build tasks while the others are sipping cocktails on the beach. I get the impression that the scheduler take its decisions in parallel resulting in all tasks ending up on the same worker.

I initially thought that the scheduler would choose a worker for the first task, then do the same for the second reevaluating the conditions until all tasks are allocated.

**Comment by deepakmohanakrishnan07** [comment]
The reason for all your task containers being placed in the same worker is due to the placement strategy you have in your configuration. The `limit-active-containers, limit-active-volumes` take priority than `volume-locality` in your case.

To distribute your workload across multiple workers, and let concourse place the containers where most of its inputs does exist, we have configured just `volume-locality` and seeing a great result. On the downside, it might add little extra time (in seconds to a minute) to stream inputs to the worker for streaming some of the inputs, but it should be negligible/acceptable step initialization delay when compared to the overall time taken by the job to complete.

**Reply by deepakmohanakrishnan07** [reply]
The documentation is clear on this https://concourse-ci.org/container-placement.html#chaining-placement-strategies
